cwl/src/main/scala/cwl/CwlExpressionCommandPart.scala

```diff
@@ -1,76 +1,79 @@
 package cwl
 
+import common.validation.ErrorOr.ErrorOr
+import common.validation.Validation._
 import cwl.CommandLineTool.CommandInputParameter
-import wom.CommandPart
 import wom.callable.RuntimeEnvironment
 import wom.expression.IoFunctionSet
 import wom.graph.LocalName
 import wom.values._
+import wom.{InstantiatedCommand, CommandPart}
 
+import scala.util.Try
+
+import scala.language.postfixOps
 
 case class CwlExpressionCommandPart(expr: Expression) extends CommandPart {
   override def instantiate(inputsMap: Map[LocalName, WomValue],
                            functions: IoFunctionSet,
                            valueMapper: (WomValue) => WomValue,
-                           runtimeEnvironment: RuntimeEnvironment ): String = {
-
-    val stringKeyMap = inputsMap.map{ case (LocalName(localName), value) => localName -> value }
+                           runtimeEnvironment: RuntimeEnvironment ): ErrorOr[InstantiatedCommand] =
+    Try {
```

Comment:
What might `throw` here, and can we wrap that small piece instead of wrapping the whole top level function here?
(also applies on the `Try`s a few times below)

Reply:
Honestly I have no idea and didn't have time to investigate.